crates/oci-distribution/src/client.rs

```diff
@@ -47,6 +47,12 @@ pub struct Client {
     client: reqwest::Client,
 }
 
+/// A source that can provide a `ClientConfig`.
```

Comment:
Nit: It might be good to expand the documentation here a bit to say something like "This is useful when writing your own configuration for an application as it can implement this trait to provide a `ClientConfig`"